test/box-luatest/downgrade_test.lua

```diff
@@ -783,3 +724,54 @@ g.test_downgrade_user_auth_history_and_last_modified = function(cg)
         check_NO_new_auth_fields()
     end)
 end
+
+-- Check that SQL CHECK and FOREIGN KEY constraints could be downgraded.
+g.test_downgrade_sql_tuple_contraints = function(cg)
+    cg.server:exec(function()
+        local _func = box.space._func
+        local _ck = box.space._ck_constraint
+        local _fk = box.space._fk_constraint
+
+        box.execute([[CREATE TABLE t(i INT PRIMARY KEY, a INT, UNIQUE(a, i));]])
+        box.execute([[CREATE TABLE t1 (i INT PRIMARY KEY, a INT);]])
+        box.execute([[ALTER TABLE t ADD CONSTRAINT c1 CHECK (a > 10);]])
+        box.execute([[ALTER TABLE t ADD CONSTRAINT c2 CHECK (abs(a) < 100);]])
+        box.execute([[ALTER TABLE t1 ADD CONSTRAINT c3 CHECK (a * i == 99);]])
+        box.execute([[ALTER TABLE t ADD CONSTRAINT f1 ]]..
+                    [[FOREIGN KEY (a) REFERENCES t;]])
+        box.execute([[ALTER TABLE t1 ADD CONSTRAINT f2 ]]..
+                    [[FOREIGN KEY (i, a) REFERENCES t(a, i);]])
+
+        local s = box.space._space.index[2]:get{'T'}
+        local s1 = box.space._space.index[2]:get{'T1'}
+        t.assert_equals(_func:get{s.flags.constraint.C1}.body, 'a > 10')
+        t.assert_equals(_func:get{s.flags.constraint.C2}.body, 'abs(a) < 100')
+        t.assert_equals(_func:get{s1.flags.constraint.C3}.body, 'a * i == 99')
+        t.assert_equals(s.flags.foreign_key.F1.space, s.id)
+        t.assert_equals(s.flags.foreign_key.F1.field, {[1] = 0})
+        t.assert_equals(s1.flags.foreign_key.F2.space, s.id)
+        t.assert_equals(s1.flags.foreign_key.F2.field, {[0] = 1, [1] = 0})
+
+        box.schema.downgrade('2.10.4')
```

Comment:
Let's use `helper.prev_version('2.11.0')` instead as in other tests. Also let's test `downgrade_issues` too.

Also let's check that the handler is not triggered if we downgrade to '2.11.0` (as in other tests).

Reply:
Fixed.